posts/fakes.md

```diff
@@ -0,0 +1,284 @@
+<meta name="id" content="7799320222051028141">
+<meta name="labels" content="testing,object oriented programming,java,microservices,domain-driven design">
+<meta name="title" content="The secret world of testing without mocking: domain-driven design, fakes, and other patterns to simplify testing in a microservices architecture">
+<meta name="description" content="How to simplify microservice testing through a few simple 
+applications of domain-driven design and in-memory test doubles (fakes).">
+
+Much has been said about mocks, the controversial, swiss-army knife of test doubles. Mocking has
+become so ubiquitous due to the popularity, flexibility, and convenience of mocking libraries like 
+Mockito, [one of the most used Java libraries in the world][mockito-popularity], that I think we may
+have forgotten there are [still more options][mocks-arent-stubs], like their older and wiser cousin,
+the humble {fake}. {Fakes} are "objects [that] actually have working implementations, but usually 
+take some shortcut which makes them not suitable for production."^[1]
+
+// TODO: maybe this focuses too much on fakes in intro
```

Comment:
I do like your previous intro better, it's more easy to read and it also gives the impression that you are speaking from experience, which you are.

Reply:
I'm thinking now maybe both intros / thesis are too narrow. I don't like the other one because it's too focused on mocking, and there are just so many essays on the internet already about that, but the point I'm trying to get to is really not talked about much. As far as subject, this intro is closer, but maybe still too narrow. I might make it about testing in general, and incorporate some of the other ideas we've talked about like production observability.

That said, I also agree the previous intro read better. Will work on this.

Reply:
I might come around to the intro last, after I've tightened up what I want to say throughout.

Reply:
Turns out I couldn't put it aside 😅 . I brought back a little of the original to your point. Let me know what you think of latest.